src/diagnostics/MOM_spatial_means.F90

```diff
@@ -220,48 +228,63 @@ function global_layer_mean(var, h, G, GV, scale, tmp_scale, unscale)
                                                                  !! units to enable the use of the reproducing sums
   real,                            optional, intent(in)  :: tmp_scale !< A temporary rescaling factor for the
-                                                         !! variable that is reversed in the return value [a A-1 ~> 1]
+                                                                 !! variable for use in the reproducing sums
+                                                                 !! that is reversed in the return value [a A-1 ~> 1],
+                                                                 !! or [b B-1 ~> 1] if unscale is also present.
   real,                            optional, intent(in)  :: unscale !< A rescaling factor for the variable [a A-1 ~> 1]
                                                                  !! that converts it back to unscaled (e.g., mks)
-                                                                 !! units to enable the use of the reproducing sums.
+                                                                 !! units to enable the use of the reproducing sums, or
+                                                                 !! a factor converting between rescaled units if
+                                                                 !! tmp_scale is also present [B A-1 ~> b a-1].
                                                                  !! Here scale and unscale are synonymous, but unscale
                                                                  !! is preferred and takes precedence.
-  real, dimension(SZK_(GV)) :: global_layer_mean  !< The mean of the variable in the arbitrary scaled [A]
-                                                  !! or unscaled [a] units of var, depending on which optional
-                                                  !! arguments are provided
+  real, dimension(SZK_(GV)) :: global_layer_mean  !< The mean of the variable in the arbitrary scaled [A ~> a]
+                                                  !! or [B ~> b] or unscaled [a] units of var, depending on which
+                                                  !! optional arguments are provided
 
   ! Local variables
-  ! In the following comments, [A] is used to indicate the arbitrary, possibly rescaled units of the
+  ! In the following comments, [A ~> a] is used to indicate the arbitrary, possibly rescaled units of the
   ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums
-  real, dimension(G%isc:G%iec,G%jsc:G%jec,SZK_(GV)) :: tmpForSumming  ! An unscaled cell integral [a m3] or [a kg]
+  real, dimension(G%isc:G%iec,G%jsc:G%jec,SZK_(GV)) :: tmpForSumming  ! An unscaled cell integral [L2 a m ~> a m3]
+                                                    ! or [L2 a kg m-2 ~> a kg] or [L2 B m ~> b m3] or [L2 B m ~> b m3]
```

Comment:
This one is extra-complicated. `weight` has two units (Boussinesq or not) and `scalefac` has three, so `tmpForSumming` has six.

|Boussinesq |`weight` |`tmp_scale` | `unscale` | `scalefac` | `tmpForSumming` |
|-|-|-|-|-|-|
|T |[L2 m ~> m3] |T |T |[B A-1 ~> b a-1] |[L2 B m ~>  b m3] |
|T |[L2 m ~> m3] |T |F |[1] |[L2 A m ~> a m3] |
|T |[L2 m ~> m3] |F |T |[a A-1 ~>1] |[L2 a m ~> a m3] |
|T |[L2 m ~> m3] |F |F |[1] |[L2 A m ~> a m3] |
|F |[L2 kg m-2 ~> kg] |T |T |[B A-1 ~> b a-1] |[L2 B kg m-2 ~>  b kg] |
|F |[L2 kg m-2 ~> kg] |T |F |[1] |[L2 A kg m-2 ~> a kg] |
|F |[L2 kg m-2 ~> kg] |F |T |[a A-1 ~>1] |[L2 a kg m-2 ~> a kg] |
|F |[L2 kg m-2 ~> kg] |F |F |[1] |[L2 A kg m-2 ~> a kg] |

Reply:
I like this table so much that I have incorporated it directly into a comment in the code.